lib/qmi/codec/device_management.ex

```diff
@@ -180,4 +182,98 @@ defmodule QMI.Codec.DeviceManagement do
        ) do
     parse_device_serial_numbers(serial_numbers, rest)
   end
+
+  @operating_modes %{
+    0x00 => :online,
+    0x01 => :low_power,
+    0x02 => :factory_test,
+    0x03 => :offline,
+    0x04 => :resetting,
+    0x05 => :shutting_down,
+    0x06 => :persistent_low_power,
+    0x07 => :mode_only_low_power,
+    0x08 => :network_test_gw
+  }
+
+  @type operating_mode ::
+          :online
+          | :low_power
+          | :factory_test
+          | :offline
+          | :resetting
+          | :shutting_down
+          | :persistent_low_power
+          | :mode_only_low_power
+          | :network_test_gw
+          | 0..7
+  @type offline_reason ::
+          :host_image_misconfiguration
+          | :pri_image_misconfiguration
+          | :pri_version_incompatible
+          | :device_memory_full
+  @type operating_mode_response :: %{
+          required(:operating_mode) => operating_mode(),
+          optional(:offline_reason) => offline_reason(),
+          optional(:hardware_controlled_mode?) => boolean()
+        }
+
+  @doc """
+  Get the operating mode of a device
+  """
+  @spec get_operating_mode() :: QMI.request()
+  def get_operating_mode() do
+    %{
+      service_id: 0x02,
+      payload: <<@get_operating_mode::16-little, 0, 0>>,
+      decode: &parse_operating_mode(%{}, &1)
+    }
+  end
+
+  defp parse_operating_mode(
+         result,
+         <<@get_operating_mode::16-little, _size::16-little, _result_tlv::7*8, 1, 1::16-little,
+           mode_num, tlvs::binary>>
+       ) do
+    mode = @operating_modes[mode_num] || {:unknown, mode_num}
```

Comment:
Curious, did hit an instance where the mode number didn't map to a known value? From what I read you accounted for them all in `@operating_modes` and so we shouldn't have an unknown mode number.

Reply:
Just being overly cautious. I guess that doesn't really help in this case and we'd prob want things to 🔥 instead